compredict/client.py

```diff
@@ -37,12 +42,20 @@ def __init__(self,
         in dashboard)
         :param passphrase: Password to the private key.
         """
-        if token is not None and len(token) != 40:
-            raise Exception("API Key is not in valid format!")
+        self.url = api.BASE_URL.format(api.API_VERSION) if url is None else url
+        self.connection = Connection(url=self.url)
+        self.token = token
+
+        if token is not None and validate:
+            self.verify_token(token)
+            self.token = token
+
+        self.token_refresh = token_refresh
+        if token is None and (username is not None and password is not None):
```

Comment:
I would make this an `elif` and add an else statement for when all token and (username or password) are None, then it is no way to proceed. so raise a ValueError.

Reply:
Alright. In this case indeed assigning token later would be redundant.